internal/config/ulimits/ulimits.go

```diff
@@ -23,25 +27,64 @@ func New() *Config {
 	}
 }
 
+// appends a ulimit to the c.Ulimits struct
+func (c *Config) appendUlimit(name string, hard, soft uint64) {
+	// This sucks, but it's the runtime-tools interface
+	c.ulimits = append(c.ulimits, Ulimit{
+		Name: name,
+		Hard: hard,
+		Soft: soft,
+	})
+}
+
+// parses a ulimit from a string config
+func (c *Config) parseUlimit(u string) (name string, hard, soft uint64, err error) {
+	ul, err := units.ParseUlimit(u)
+	if err != nil {
+		return "", 0, 0, fmt.Errorf("unrecognized ulimit %s: %v", u, err)
+	}
+	rl, err := ul.GetRlimit()
+	if err != nil {
+		return "", 0, 0, err
+	}
+	return "RLIMIT_" + strings.ToUpper(ul.Name), rl.Hard, rl.Soft, nil
+}
+
 func (c *Config) LoadUlimits(ulimits []string) error {
 	// Process and initialize ulimits at cri-o start up, so crio fails early if
 	// its misconfigured. After this, we can always refer to config.Ulimits() to get
 	// the configured Ulimits
 	for _, u := range ulimits {
-		ul, err := units.ParseUlimit(u)
+		name, hard, soft, err := c.parseUlimit(u)
 		if err != nil {
-			return fmt.Errorf("unrecognized ulimit %s: %v", u, err)
+			return err
 		}
-		rl, err := ul.GetRlimit()
+		c.appendUlimit(name, hard, soft)
+	}
+	return nil
+}
+
+// Loads ulimits from a profile specified in profilepath
+func (c *Config) LoadUlimitsFromProfile(profilePath string, g *generate.Generator) error {
+	file, err := os.Open(profilePath)
+	if err != nil {
+		return fmt.Errorf("failed to open %s, %v", profilePath, err)
+
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		logrus.Info("setting ulimit: ", scanner.Text())
+		name, hard, soft, err := c.parseUlimit(scanner.Text())
```

Comment:
this naturally limits it so we can only limit one thing. what if a user wants to pass multiple ulimits?

Reply:
I'm a little confused, 
this allows you to set multiple ulimits when you give a profile with multiple ulimits
https://github.com/cri-o/cri-o/pull/4044/files#diff-5100eebf92d097811018ce4f7067a915R17

Do you mean if a user wants to use multiple profiles ?

Reply:
I must be misreading the code! my mistake, this is working as I'd hope it would :smiley: